### ISSUE TITLE
Full dynamic import using function argument destructuring breaks on production builds

### ISSUE THREAD

**Issue by kamikazePT** [gregberge/loadable-components#514]
## 🐛 Bug Report

For example on a create-react-app, if i use function argument destructuring on a full dynamic import, the props are seen as undefined, only on production build ( probably minifier optimizations going on... )

What's funny is that if i use "lazy" instead of "loadable" it seems to work...

## To Reproduce

For example having a Sidebar for each type of user role ( normal and admin for example )
const Sidebar = loadable(({ role }) => import(`./${role}`));

myRole being "admin"

const Layout => (
 ....
 <Sidebar role={myRole}/>
 ...
)

## Expected behavior

if a component called admin.js or a folder admin with index.js exists on that path, it should be loaded, and in fact... on development mode, it is loaded...

on production build "role" inside the import function is undefined.

If i change this:
const Sidebar = loadable(({ role }) => import(`./${role}`));
to this:
const Sidebar = loadable((props) => import(`./${props.role}`));

It works.... using lazy instead of loadable works both ways

## Link to repl or repo (highly encouraged)

Please provide a minimal repository on GitHub.

Issues without a reproduction link are likely to stall.

## Run `npx envinfo --system --binaries --npmPackages @loadable/component,@loadable/server,@loadable/webpack-plugin,@loadable/babel-plugin --markdown --clipboard`

Paste the results here:

```bash

```

I'll add the reproduction link soon as I have more time, sorry

**Comment by theKashey** [comment]
> using lazy instead of loadable works both ways

lazy does not provide any `props` to the importer. So, no, it would not work with `lazy` in any form.
However, speaking of this "bug" - it's not related to Loadable(I think that's Babel), and could not be fixed from Loadable side, but we would try.

Could you change both versions to something like
```js
const Sidebar = loadable(({ role }) => {
   console.log(arguments);
   console.log(role);
   return import(./${role})
});
```
And post here result for both versions.

**Comment by kamikazePT** [comment]
@theKashey I have something like this to route based on roles and it's working correctly, so you telling me It would never work, comes as a surprise o.o
```
import { lazy } from '@loadable/component';

const Page = lazy(({ role }) => import(`./${role}`));
```
I believe its related to the minifier not babel per say ( i believe under the hood CRA uses terser )

either way, the results:

argument destructuring
![Screenshot from 2020-01-15 01-50-36](https://user-images.githubusercontent.com/6997763/72398204-2fee5200-373a-11ea-8eee-c5fceb3b77ec.png)

normal props
![Screenshot from 2020-01-15 01-54-37](https://user-images.githubusercontent.com/6997763/72398215-367cc980-373a-11ea-8cf2-da0b2aa7dc40.png)

**Comment by theKashey** [comment]
- `lazy` == `React.lazy` 😅
- `arguments` seems to display something wrong. Look like these are "module" variables.

Anyway, the best way to tackle this problem is to add your case to babel plugin tests cases.

**Comment by theKashey** [comment]
Partially related to #538
Would be resolved by #483
